mqtt3/1.0/modules/ROOT/edits/mqtt3-connector-client-id.adoc

```diff
@@ -0,0 +1,355 @@
+= Set a Client ID with MQTT Connector
+
+Client IDs are important for MQTT because they identify a connection to an MQTT broker. Moreover, they identify a specific device or client,
+and it can be useful in terms of traceability to set a client ID that clearly identifies a device's connection to the broker.
+
+MQTT brokers usually implement a mechanism that closes an old connection if a new connection request with the same client ID as the existing (old) connection is received and then accepts the new connection. MQTT brokers employ this mechanism to prevent situations such as TCP connections reaching a half-open or invalid state due to a crash in one of the communication ends, which is undesirable. The broker assumes the new incoming connection to be valid, closes the older connection, and continues using the newly established connection.
+
+The client takeover mechanism can become a problem if you want to deploy an application in a cluster, and still provide an explicit client ID that identifies the client. Anypoint Connector for MQTT (MQTT Connector) provides client ID generators that help you set a static client ID, set a client ID with a dynamic component to adjust to different deployment scenarios, or define your custom client ID generator.
```

Comment:
Is the "client takeover mechanism" the situation described just above? If so, consider working that phrase into the prev. paragraph.

Reply:
This paragraph confuses me a bit, because we have not defined explicit, static, or dynamic IDs.
Also, we have not made a distinction between MQTT ID generators and custom generators.
And...how does the first sentence, about the cluster problem, relate to the second, about different types of client IDs and generators?

Reply:
Finally...the focus on this introduction to the three tasks that follow should clearly identify those three tasks and why the audience wants/needs to perform them:

- Configure custom expression generator (what is an expression? Not mentioned in intro, just ID....)
- Configure random suffix generator (not mentioned at all in the intro)
- Configure custom ID generator (how is this diff. from the first task?)

Reply:
The intro does not clarify that there are three generators that you can configure, vs. whatever MQTT does without your configuration.